### ISSUE TITLE
Need a 5B backbone to close the control loop: 14B is ~360x too slow on gfx1151

### ISSUE THREAD

**Issue by raikonenfnu** [raikonenfnu/dexter-labs#1]
## Problem

The released DreamZero checkpoints cannot close a control loop on gfx1151, and
no amount of kernel work fixes it. A 5B backbone can — but no action-trained 5B
checkpoint exists, so we would have to produce one.

## Why the 14B can't work here

Measured on this box (Strix Halo, 40 CU, 32 GB unified, 233 GB/s, 29.5 TFLOP/s bf16):

| | |
|---|---|
| DreamZero-DROID 14B, int4 | **24 s** / control step |
| compute floor at 100% of peak | **6.6 s** / control step |
| DROID control period (15 Hz) | **66.7 ms** |

That is ~360x too slow, and the floor is arithmetic rather than an
implementation gap: `2 x 13.6e9 params x 1785 tokens = 48.7 TFLOP` per denoising
step, x4 steps, at 29.5 TFLOP/s. Even a perfect kernel leaves ~0.15 Hz.

Action chunking does not rescue it. Upstream executes `open_loop_horizon=8`
actions before re-querying, which at 15 Hz is 533 ms of queued motion — the
entire budget available to hide inference behind. 24 s against 533 ms means the
arm drains its queue in half a second and freezes for the remaining ~23.5 s.
Pipelining (worth 9.0 -> 14.5 Hz at 5B, see below) needs inference to roughly
fit inside the chunk duration; here it is off by ~15x.

int4 is also not optional: bf16 is 27.3 GB and does not fit in 32 GB alongside
the OS. At 1785 tokens the workload sits past this machine's 126 FLOP/byte
balance point, so it is compute-bound — quantisation buys *capacity*, not
latency, and the int4 GEMM currently runs ~1.8x slower than dense. There is no
configuration where the 14B both fits and runs fast here.

## Why 5B changes the picture — 57x, not 2.7x

| | 14B (released) | 5B (recipe only) |
|---|---|---|
| params | 13.6B | 5.1B |
| video | 352x640, VAE 8x | 160x320, VAE **16x** |
| tokens/frame | 880 | 50 |
| frames/block | 2 | 1 |
| **tokens/step** | **1785** | **83** |
| FLOP/denoise step | 48.7 T | 0.85 T |

The parameter count is only 2.7x. The real win is the **21x token drop**, from
the smaller resolution *and* Wan2.2's 16x VAE *and* one frame per block.

Built to the wan22 config shapes (30 layers, dim 3072, 24 heads, ffn 14336 —
architecture-accurate, randomly initialised, so timing is shape-determined):

```
bf16   10.35 GB   361 ms / control step
```

361 ms fits inside the 533 ms an 8-action chunk buys, so the chunk scheduler in
`dexter/demo/robot.py` closes the loop:

```
blocking       9.0 Hz   stall 47.6%   misses 19/150
pipelined/pf8 14.5 Hz   stall  3.9%   misses  1/150   (15 Hz target)
```

## The blocker: no 5B checkpoint exists

DreamZero ships the 5B as a **training recipe**, not weights:

- `docs/WAN22_BACKBONE.md`
- `configs/model/dreamzero/action_head/wan_flow_matching_action_tf_wan22.yaml`
- `scripts/train/droid_training_wan22.sh`

Both released checkpoints (`GEAR-Dreams/DreamZero-DROID`,
`GEAR-Dreams/DreamZero-AgiBot`) use the same 14B backbone —
`dim=5120, 40 layers, 40 heads, ffn=13824, frame_seqlen=880, in_dim=36`. AgiBot
is a different *embodiment*, not a smaller model.

`Wan-AI/Wan2.2-TI2V-5B` is a pure video model — no action head, no action
register, no policy weights.

Relevant upstream threads (context, not action items for us):

- [dreamzero0/dreamzero#31](https://github.com/dreamzero0/dreamzero/issues/31) —
  request for a 5B checkpoint. Closed; maintainer says WAN2.2-5B *support* was
  merged but released no weights. Two users still asking.
- [#79](https://github.com/dreamzero0/dreamzero/issues/79) — `droid_training_wan22.sh`
  uses `train_architecture=lora` while the paper describes full fine-tuning for
  pretraining; reporter OOMs attempting full train on a 96 GB H20.
- [#73](https://github.com/dreamzero0/dreamzero/issues/73) — why the Wan2.2
  backbone injects first-frame CLIP via cross-attention.

## Options

1. **Train DreamZero-5B on DROID.** Recipe exists, preprocessed data is released
   (`GEAR-Dreams/DreamZero-DROID-Data`). Per #79 the shipped script is LoRA, and
   full training OOMs on 96 GB — so this is not a weekend job on 32 GB, and
   probably not feasible on this hardware at all.
2. **LoRA post-train the 5B** from the Wan2.2 base. Cheaper, but the base has no
   action head — the action encoder/decoder and register would train from
   scratch, which is more than a LoRA.
3. **Ask upstream to release the 5B checkpoint**, with the measured evidence
   above. #31 suggests appetite exists but no plan.
4. **Use a different small policy** (PushT diffusion, ACT, pi0) for a robot demo
   today, and keep DreamZero as the offline world-model demo.

## Engine work this does *not* block

Everything already built — the gfx1151 kernels, int4 streaming loader, chunk
scheduler, perception stack — applies unchanged to a 5B checkpoint if one
appears. The only 5B-specific engine change needed is `latent_height/width`
(5x10 vs 22x40), already in `WAN22_TI2V_5B`.

Separately worth doing regardless: the int4 GEMM is ~1.8x off dense
(dequantisation codegen, see README), which is 1.8x on *any* backbone.
